src/Main.cpp

```diff
@@ -268,3 +295,50 @@ static void editSettings() {
 static void showAbout() {
 	ShowAboutDialog((HINSTANCE)_hModule, MAKEINTRESOURCE(IDD_ABOUTDLG), nppData._nppHandle);
 }
+
+static void setLanguage() {
+	std::string language = ShowLanguageDialog((HINSTANCE)_hModule, MAKEINTRESOURCE(IDD_LANGUAGEDLG), nppData._nppHandle, config);
+	if (!language.empty()) {
+		wchar_t fileName[MAX_PATH] = { 0 };
+		SendMessage(nppData._nppHandle, NPPM_GETFULLCURRENTPATH, 0, (LPARAM)fileName);
+		bufferLanguages[UTF8FromString(fileName).c_str()] = language;
+		ScintillaGateway editor(GetCurrentScintilla());
+		SetLexer(editor, language);
+	}
+}
+
+static void editLanguageDefinition() {
+	std::string language = ShowLanguageDialog((HINSTANCE)_hModule, MAKEINTRESOURCE(IDD_LANGUAGEDLG), nppData._nppHandle, config);
+	if (!language.empty()) {
+		wchar_t config_dir[MAX_PATH] = { 0 };
+		SendMessage(nppData._nppHandle, NPPM_GETPLUGINSCONFIGDIR, MAX_PATH, (LPARAM)config_dir);
+		wcscat_s(config_dir, MAX_PATH, L"\\Scintillua++\\");
+		wcscat_s(config_dir, MAX_PATH, StringFromUTF8(language).c_str());
+		wcscat_s(config_dir, MAX_PATH, L".lua");
+		SendMessage(nppData._nppHandle, NPPM_DOOPEN, 0, (LPARAM)config_dir);
+	}
+}
+
+static void createNewLanguageDefinition() {
+	wchar_t config_dir[MAX_PATH] = { 0 };
+	SendMessage(nppData._nppHandle, NPPM_GETPLUGINSCONFIGDIR, MAX_PATH, (LPARAM)config_dir);
+	wcscat_s(config_dir, MAX_PATH, L"\\Scintillua++\\template.txt");
+	std::ifstream t(config_dir);
+	std::string str;
+
+	t.seekg(0, std::ios::end);   
+	str.reserve(t.tellg());
```

Comment:
In 32bit build this gives:

> warning C4244: 'argument' : conversion from 'std::streamoff' to 'unsigned int', possible loss of data

Not a huge problem but the release builds treat warnings as errors.